### DISCUSSION TITLE
Steps.[step_id].outcome == 'failure' does not work

### DISCUSSION THREAD

**Discussion by aqeelat** [community/community#26732] [Actions]
<p>Hi GitHub,</p>
<p>This is a snippet from my job:</p>
<pre><code class="lang-auto">- name: Run Test With Coverage # running tests
    id: test_step
    run: pytest
- name: Dump steps context
    if: ${{ always() }}
    env:
        STEPS_CONTEXT: ${{ toJson(steps) }}
    run: echo "$STEPS_CONTEXT"
- name: Re Run Failed Tests
    if: ${{ steps.test_step.outcome == 'failure' }}
    run: LOG_LEVEL=info pytest --lf
</code></pre>
<p>The result of the context dump is:</p>
<pre><code class="lang-auto">{
  "eba14a9f4c4e4c77b1c5cb59ba84c279": {
    "outputs": {},
    "outcome": "success",
    "conclusion": "success"
  },
  "172ab47f570540e084d249e62f611c31": {
    "outputs": {},
    "outcome": "success",
    "conclusion": "success"
  },
  "test_step": {
    "outputs": {},
    "outcome": "failure",
    "conclusion": "failure"
  }
}
</code></pre>
<p>However, the third step (<em>re run failed tests</em>) won’t execute.</p>
<ol>
<li>Why does the dump only contains 3 steps, even though I have about 10+ steps?</li>
<li>What did I do wrong, and how I can I the steps execute?</li>
</ol>
<p>Thanks,<br>
Abdullah</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="AqeelAT" data-post="1" data-topic="215118">
<div class="title">
<div class="quote-controls"></div>
 AqeelAT:</div>
<blockquote>
<p>However, the third step (<em>re run failed tests</em>) won’t execute.</p>
</blockquote>
</aside>
<p>The <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#job-status-check-functions">documentation on status check functions</a> says:</p>
<blockquote>
<p>You can use the following status check functions as expressions in <code>if</code> conditionals. A default status check of <code>success()</code> is applied unless you include one of these functions.</p>
</blockquote>
<p>That’s why, you’ll need to add one of the other status check functions (I assume <code>failure()</code>) to the <code>if</code> condition.</p>

**Comment by aqeelat** [comment]
<p>Thank You!<br>
I changed my if statement to: <code>if: ${{ failure() &amp;&amp; steps.test_step.outcome == 'failure' }}</code> and it worked. I with this was in the documentation because it was frustrating to figure out.</p>
<p>I had used failure before, but I added black and isort and I wanted them to fail the workflow if the PR wasn’t compliant with them, but I didn’t want the rerun failed tests step to execute (which would’ve happened with <code>if: {{ failure() }}</code></p>
